### ISSUE TITLE
defaultdict generic value parameter is inferred from constructor if parameter is a type variable

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1084]
**Describe the bug**
If a `defaultdict` is declared with a value whose type is a generic and its parameter is a type variable, the parameter is ignored and the type is inferred from the `defaultdict` constructor.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `bar` would be of type `defaultdict[str, list[T]]`; that the value of `baz` would be incompatible with its declared type.

**Screenshots or Code**
```python
# pyright: strict

from typing import Type, TypeVar

from collections import defaultdict

T = TypeVar('T')

def foo(value: Type[T]) -> None:
    bar: 'defaultdict[str, list[T]]' = defaultdict(list)
    reveal_type(bar)  # Type of "baz" is "defaultdict[str, list[Unknown]]"; should be list[T]

    baz: 'defaultdict[str, list[T]]' = defaultdict(list[int])
    reveal_type(baz)  # Type of "baz" is "defaultdict[str, list[int]]"; should error

```

**VS Code extension or command-line**
Extension, 1.1.77

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.87, which I just published. It will also be included in the next release of pylance.
